<ui version="4.0" >
 <class>TestBed</class>
 <widget class="QMainWindow" name="TestBed" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>601</width>
    <height>376</height>
   </rect>
  </property>
  <property name="palette" >
   <palette>
    <active>
     <colorrole role="WindowText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>128</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>128</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Light" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>76</red>
        <green>76</green>
        <blue>76</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Midlight" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>63</red>
        <green>63</green>
        <blue>63</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Dark" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>25</red>
        <green>25</green>
        <blue>25</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Mid" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>34</red>
        <green>34</green>
        <blue>34</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>128</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>51</red>
        <green>51</green>
        <blue>51</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>76</red>
        <green>76</green>
        <blue>76</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Shadow" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="AlternateBase" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>25</red>
        <green>25</green>
        <blue>25</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipBase" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>255</green>
        <blue>220</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="WindowText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>128</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>128</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Light" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>76</red>
        <green>76</green>
        <blue>76</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Midlight" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>63</red>
        <green>63</green>
        <blue>63</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Dark" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>25</red>
        <green>25</green>
        <blue>25</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Mid" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>34</red>
        <green>34</green>
        <blue>34</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>128</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>51</red>
        <green>51</green>
        <blue>51</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>76</red>
        <green>76</green>
        <blue>76</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Shadow" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="AlternateBase" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>25</red>
        <green>25</green>
        <blue>25</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipBase" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>255</green>
        <blue>220</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="WindowText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>25</red>
        <green>25</green>
        <blue>25</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>128</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Light" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>76</red>
        <green>76</green>
        <blue>76</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Midlight" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>63</red>
        <green>63</green>
        <blue>63</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Dark" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>25</red>
        <green>25</green>
        <blue>25</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Mid" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>34</red>
        <green>34</green>
        <blue>34</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>25</red>
        <green>25</green>
        <blue>25</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>25</red>
        <green>25</green>
        <blue>25</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>76</red>
        <green>76</green>
        <blue>76</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>76</red>
        <green>76</green>
        <blue>76</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Shadow" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="AlternateBase" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>51</red>
        <green>51</green>
        <blue>51</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipBase" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>255</red>
        <green>255</green>
        <blue>220</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipText" >
      <brush brushstyle="SolidPattern" >
       <color alpha="255" >
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="windowTitle" >
   <string>TestBed</string>
  </property>
  <property name="autoFillBackground" >
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget" >
   <layout class="QHBoxLayout" name="horizontalLayout_8" >
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2" >
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" >
        <item>
         <widget class="SideCar::GUI::LED" name="ledA_" >
          <property name="minimumSize" >
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="maximumSize" >
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="toolTip" >
           <string>Binary Led</string>
          </property>
          <property name="whatsThis" >
           <string>Led widget</string>
          </property>
          <property name="color" >
           <enum>SideCar::GUI::LED::kRed</enum>
          </property>
          <property name="diameter" >
           <number>16</number>
          </property>
          <property name="value" stdset="0" >
           <bool>true</bool>
          </property>
          <property name="pending" stdset="0" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="actionA_" >
          <property name="text" >
           <string>Action A</string>
          </property>
          <property name="iconSize" >
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
          <property name="checkable" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7" >
        <item>
         <widget class="SideCar::GUI::LED" name="ledB_" >
          <property name="maximumSize" >
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
          <property name="toolTip" >
           <string>Binary Led</string>
          </property>
          <property name="whatsThis" >
           <string>Led widget</string>
          </property>
          <property name="color" >
           <enum>SideCar::GUI::LED::kYellow</enum>
          </property>
          <property name="value" stdset="0" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="actionB_" >
          <property name="text" >
           <string>Action B</string>
          </property>
          <property name="checkable" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3" >
        <item>
         <widget class="SideCar::GUI::LED" name="ledC_" >
          <property name="minimumSize" >
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="maximumSize" >
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="toolTip" >
           <string>Binary Led</string>
          </property>
          <property name="whatsThis" >
           <string>Led widget</string>
          </property>
          <property name="color" >
           <enum>SideCar::GUI::LED::kGreen</enum>
          </property>
          <property name="diameter" >
           <number>32</number>
          </property>
          <property name="value" stdset="0" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="actionC_" >
          <property name="text" >
           <string>Action C</string>
          </property>
          <property name="checkable" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4" >
        <item>
         <widget class="SideCar::GUI::LED" name="ledD_" >
          <property name="toolTip" >
           <string>Binary Led</string>
          </property>
          <property name="whatsThis" >
           <string>Led widget</string>
          </property>
          <property name="color" >
           <enum>SideCar::GUI::LED::kGrey</enum>
          </property>
          <property name="diameter" >
           <number>64</number>
          </property>
          <property name="value" stdset="0" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="actionD_" >
          <property name="text" >
           <string>Action D</string>
          </property>
          <property name="checkable" >
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_3" >
        <property name="orientation" >
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout" >
      <item>
       <widget class="QGroupBox" name="mode_" >
        <property name="title" >
         <string>Mode</string>
        </property>
        <property name="flat" >
         <bool>false</bool>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_6" >
         <item>
          <widget class="QRadioButton" name="modeA_" >
           <property name="text" >
            <string>Mode A</string>
           </property>
           <property name="checked" >
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="modeB_" >
           <property name="text" >
            <string>Mode B</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2" >
        <item>
         <widget class="QLabel" name="label" >
          <property name="text" >
           <string>Choice</string>
          </property>
          <property name="alignment" >
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="choice_" >
          <item>
           <property name="text" >
            <string>Choice 1</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>Choice 2</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>Choice 3</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>Choice 4</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>Choice 5</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5" >
        <item>
         <widget class="QLCDNumber" name="dialValue_" >
          <property name="autoFillBackground" >
           <bool>false</bool>
          </property>
          <property name="frameShape" >
           <enum>QFrame::Box</enum>
          </property>
          <property name="frameShadow" >
           <enum>QFrame::Raised</enum>
          </property>
          <property name="smallDecimalPoint" >
           <bool>false</bool>
          </property>
          <property name="digitCount" >
           <number>2</number>
          </property>
          <property name="segmentStyle" >
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDial" name="dial_" >
          <property name="minimumSize" >
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="autoFillBackground" >
           <bool>false</bool>
          </property>
          <property name="wrapping" >
           <bool>false</bool>
          </property>
          <property name="notchTarget" >
           <double>10.000000000000000</double>
          </property>
          <property name="notchesVisible" >
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_2" >
        <property name="orientation" >
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>17</width>
          <height>37</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3" >
      <item>
       <widget class="SideCar::GUI::Potentiometer" name="pot" >
        <property name="value" >
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="SideCar::GUI::Potentiometer" name="pot_2" >
        <property name="maximum" >
         <double>100.000000000000000</double>
        </property>
        <property name="value" >
         <double>0.000000000000000</double>
        </property>
        <property name="step" >
         <double>20.000000000000000</double>
        </property>
        <property name="tickCount" >
         <number>5</number>
        </property>
        <property name="startColor" >
         <color>
          <red>0</red>
          <green>255</green>
          <blue>0</blue>
         </color>
        </property>
        <property name="endColor" >
         <color>
          <red>0</red>
          <green>255</green>
          <blue>0</blue>
         </color>
        </property>
        <property name="showScrews" >
         <bool>true</bool>
        </property>
        <property name="showSweep" >
         <bool>false</bool>
        </property>
        <property name="showValue" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer" >
        <property name="orientation" >
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>601</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar" />
 </widget>
 <customwidgets>
  <customwidget>
   <class>SideCar::GUI::LED</class>
   <extends>QWidget</extends>
   <header>GUI/LED.h</header>
  </customwidget>
  <customwidget>
   <class>SideCar::GUI::Potentiometer</class>
   <extends>QWidget</extends>
   <header>GUI/Potentiometer.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../guiIcons.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>dial_</sender>
   <signal>valueChanged(int)</signal>
   <receiver>dialValue_</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>316</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel" >
     <x>206</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionB_</sender>
   <signal>clicked()</signal>
   <receiver>ledB_</receiver>
   <slot>cycleColors()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>89</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel" >
     <x>29</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionA_</sender>
   <signal>clicked()</signal>
   <receiver>ledA_</receiver>
   <slot>cycleColors()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>109</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel" >
     <x>28</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionC_</sender>
   <signal>clicked()</signal>
   <receiver>ledC_</receiver>
   <slot>cycleColors()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>126</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel" >
     <x>26</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionD_</sender>
   <signal>clicked()</signal>
   <receiver>ledD_</receiver>
   <slot>cycleColors()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>100</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel" >
     <x>19</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
